### PR TITLE
feature: Switch to new Bloop Rifle in the original Bloop

### PR DESCRIPTION
**Pull request by tgodzik**
Supersedes https://github.com/scalameta/metals/pull/4103

### CODE REVIEW

**Comment by tgodzik** [comment]
Bloop PR here https://github.com/scalacenter/bloop/pull/2355

**Comment by tgodzik** [comment]
This should be good to review, though I see some issue with the test script hanging so I might need to remove it at some point.

**Review by kasiaMarek — APPROVED** [review]
Should we also update possible versions jvm versions for Metals in VSCode plugin to drop 11?
Also do we want to delete the old `.bloop` config file at a point? (maybe suggest the user deleting it)

File: metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfig.scala [hunk 1]
```diff
@@ -87,6 +88,10 @@ final case class MetalsServerConfig(
     bloopPort: Option[Int] = Option(System.getProperty("metals.bloop-port"))
       .filter(_.forall(Character.isDigit(_)))
       .map(_.toInt),
+    bloopDirectory: Option[AbsolutePath] =
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfig.scala]: Removed bloopDirectory altogether, I am not sure if we would want it to be used at any point. We can add it later if there is a need, For now, I would need to figure out if it's working and when it's working

**Pull request merged**